src/libstore/derivations.hh

```diff
@@ -289,6 +362,8 @@ struct BasicDerivation
     StringPairs env;
     std::string name;
 
+    DerivationOptions options;
```

Comment:
It seems that having this field in `BasicDerivation` "denormalizes" it, since information like the value of `__impureEnvVars` is now stored in both `drv.env` *and* `drv.options.impureEnvVars`, which could get out of sync and would therefore be error-prone to use. Wouldn't it be better to have methods like `BasicDerivation::{get,set}ImpureEnvVars()`?

Reply:
@edolstra the goal is, via the JSON format, to allow that stuff to not exist in the env. Then only the legacy decoding results looks like denormalization — but actually I prefer the perspective that there is no denormalization and rather the legacy format simply lacks the expressive power to set one without setting the other.

Does that make sense? It is some subtle philosophical perspective-shifting.

Reply:
Philisophical doesn't do it justice. It's anticipating a design improvement.

Reply:
Yeah philosophical before the new feature (better JSON format *as input*), practical with it.